evmap/src/write.rs

```diff
@@ -542,13 +308,23 @@ where
             (k, vs.user_friendly())
         })
     }
+}
 
+unsafe impl<K, V, M, S> Absorb<Operation<K, V, M>> for Inner<K, V, M, S>
+where
+    K: Eq + Hash + Clone,
+    S: BuildHasher + Clone,
+    V: Eq + Hash + ShallowCopy,
+    M: 'static + Clone,
+{
     /// Apply ops in such a way that no values are dropped, only forgotten
-    fn apply_first(
-        inner: &mut Inner<K, ManuallyDrop<V>, M, S>,
-        op: &mut Operation<K, V>,
-        hasher: &S,
-    ) {
+    fn absorb_first(&mut self, op: &mut Operation<K, V, M>, other: &Self) {
+        // Make sure that no methods below drop values since we're only operating on the first
+        // shallow copy of each value.
+        //
+        // Safety: ManuallyDrop<T> has the same layout as T.
+        let inner: &mut Inner<K, ManuallyDrop<V>, M, S> = unsafe { mem::transmute(self) };
```

Comment:
```suggestion
        let inner = unsafe { &mut *(self as *mut Inner<K, V, M, S> as *mut Inner<K, ManuallyDrop<V>, M, S>) };
```
Listed as one of the [alternatives](https://doc.rust-lang.org/std/mem/fn.transmute.html#alternatives) to transmute.